src/main/java/chatterbot/data/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package chatterbot.data;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + modifyBy(by) + ")";
+    }
+
```

Comment:
I'm not sure what this method is supposed to do based on the name. Is it to convert the "by" string to a LocalDate? If it is, you can consider changing the name to something like `convertByToLocalDate`